ConfigSpace/hyperparameters.pyx

```diff
@@ -1882,31 +1905,46 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
         rs: np.random.RandomState,
         number: int = 4,
         transform: bool = False,
-    ) -> List[Union[np.ndarray, float, int]]:
-        neighbors = []  # type: List[Union[np.ndarray, float, int]]
+    ) -> List[int]:
+        neighbors: set[int] = set()
+        stepsize = self.q if self.q is not None else 1
+        bounded = self.lower is not None
+
+        center = self._transform(value)
+
         while len(neighbors) < number:
-            rejected = True
-            iteration = 0
-            while rejected:
-                iteration += 1
-                new_value = rs.normal(value, self.sigma)
-                int_value = self._transform(value)
-                new_int_value = self._transform(new_value)
-
-                if self.lower is not None and self.upper is not None:
-                    int_value = min(max(int_value, self.lower), self.upper)
-                    new_int_value = min(max(new_int_value, self.lower), self.upper)
-
-                if int_value != new_int_value:
-                    rejected = False
-                elif iteration > 100000:
-                    raise ValueError('Probably caught in an infinite loop.')
+            possible_neighbor = self._sample(rs)
```

Comment:
This does no longer sample around the current value and thus thereby change the behavior of SMAC etc. Could you please change this? Maybe we can use a similar logic as above (except that we replace the truncated normal by a regular normal distribution)?

Reply:
Whoops, good catch, I had it like that before but realised we need different distributions depending on if it's `bounded` or not. Thought I could sneak it some re-use but this ignore `value` will fix.